test/evp_test.c

```diff
@@ -1516,7 +1516,7 @@ static int mac_test_run_mac(EVP_TEST *t)
         t->err = "MAC_CREATE_ERROR";
         goto err;
     }
-    if (fips_provider_version_gt(libctx, 3, 0, 12))
+    if (fips_provider_version_gt(libctx, 3, 1, 4))
```

Comment:
Shouldn't this rather be something like this:

```suggestion
    if ((fips_provider_version_gt(libctx, 3, 0, 12) && fips_provider_version_lt(libctx, 3, 1, 0))
        || fips_provider_version_gt(libctx, 3, 1, 4))
```

Maybe we need a new helper function for that kind of construct which I expect could become common.

Reply:
fixup pushed